### ISSUE TITLE
SplitPolygonExample.cs does split, but only returns one part of the result

### ISSUE THREAD

**Issue by chau-intl** [NetTopologySuite/NetTopologySuite#114]
Looking a the samples I have found the `SplitPolygonExample.cs` which simple splits a polygon using a line. 
![nts_split_polygon_bug](https://cloud.githubusercontent.com/assets/6311686/17327099/ffa74950-58b5-11e6-80d2-b1e16fdcee8d.JPG)

At the moment it seems like something has changed in `NetTopologySuite.Operation.Polygonize`, since the split using the line cuts the polygon but only return the left hand part. I would expect both parts to be returned.

@FObermaier do you know something about this?

**Comment by DGuidi** [comment]
can be related to #87? [other changes](https://github.com/NetTopologySuite/NetTopologySuite/tree/master/NetTopologySuite/Operation/Polygonize) are at least one year old...

**Comment by chau-intl** [comment]
I haven't tried the sample before, so I cannot tell when (or if?) it actually worked.

**Comment by FObermaier** [comment]
This example does not work with JTS either. I'm sure it did work at the time I added it.

**Comment by DGuidi** [comment]
> This example does not work with current JTS either

So we can close this issue. thanks @chau-intl for this report.

**Comment by FObermaier** [comment]
I think it broke with cd319e66 or one of the subsequent.
The new algorithm tries to take advantage of the knowing which rings are adjacent to each other. If one of the adjacent rings is included in the result set, all others are not. This is valid as long as the linework causing adjacency is not the line that shall split the polygon.
This is why we only get one.

Maybe @dr-jts has a clue on how to handle that.

**Comment by am2222** [comment]
Any Idea how to fix this problem?I tried to use difference but it returns an empty polygon

**Comment by jholt456** [comment]
I just hit this issue also. Sample I wrote a bit ago using 1.13.3.2 worked correctly, new project using 1.14.0 gave unexpected results. I dropped back to 1.13.3.2 for now.

**v1.13.3.2 Result:**
![nts-poly-1 13 3 2](https://cloud.githubusercontent.com/assets/647656/17727096/9c64cecc-641c-11e6-8a27-5b21b2ab2e73.png)

**v1.14.0 Result:**
![nts-poly-1 14](https://cloud.githubusercontent.com/assets/647656/17727104/b257fb78-641c-11e6-856b-063ec89030dc.png)

**Comment by FObermaier** [comment]
@jholt456 Could you post a unit test for the sample pictures?
I have a broad idea how to fix this issue.

**Comment by jholt456** [comment]
Sure thing @FObermaier.  The images were captured from an application I am working on, but this test replicates the issue.  It returns 64 polys on v1.13.3.2, and 32 on v1.14.  Please let me know if I can send anything else over.

```c# 
[TestMethod]
public void Should_Find_Correct_Number_Of_Polygons_From_Lines()
{
    var paths = new List<IGeometry>();
           
    for (int x = 1; x < 10; x++)
    {
        var startPoint = new Coordinate(x * 40, 30);
        var endPoint = new Coordinate(x * 40, 440);
        paths.Add(new LineString(new[] { startPoint, endPoint }));
    }
            
    for (int y = 1; y < 10; y++)
    {
        var startPoint = new Coordinate(30, y * 40);
        var endPoint = new Coordinate(450, y * 40);

        paths.Add(new LineString(new[] { startPoint, endPoint }));
    }

    var noder = new NetTopologySuite.Noding.Snapround.GeometryNoder(new PrecisionModel(1.0d));

    var geomFactory = GeometryFactory.Default;
    var nodedLines = noder.Node(paths);

    var nodedDedupedLinework = geomFactory.BuildGeometry(nodedLines.ToArray()).Union();

    Polygonizer polygonizer = new Polygonizer();
    polygonizer.Add(nodedDedupedLinework);

    var polygons = polygonizer.GetPolygons();
    Assert.AreEqual(64, polygons.Count);
}
```

**Comment by FObermaier** [comment]
Your issue can be resolved using the `new Polygonizer(false)` constructor.

**Comment by FObermaier** [comment]
In fact it turned out that the split polygon example will continue to work when using the constructor mentioned above. Sorry!

**Comment by am2222** [comment]
@jholt456 a not related question, Can you say what library are you using to draw shapes?thanks

**Comment by jholt456** [comment]
@am2222 Not using a library in the images I posted due to interactivity requirements for this project.  I use NTS for the heavy lifting behind the scenes, but the view just binds an ItemsControl to a Shapes collection on the view model.  Each shape type has a DataTemplate to specify how it will be rendered. In the case of a Polygon, the data template renders a standard WPF polygon, all points from the shape (used as drag handles), and a label. 

In the past I have used the [WpfPathGeometryWriter](https://github.com/NetTopologySuite/NetTopologySuite/blob/dc3cfd98f42d4c74f6e1fd8c25e644748cd29ae6/NetTopologySuite.Windows.Media/WpfPathGeometryWriter.cs) provided by NTS, and it worked great for quickly displaying shapes. 

Hope that helps, let me know if I can give you anymore info.
